server/lib/tuist/qa.ex

```diff
@@ -110,6 +115,93 @@ defmodule Tuist.QA do
     "#{String.downcase(account_handle)}/#{String.downcase(project_handle)}/qa/screenshots/#{qa_run_id}/#{file_name}.png"
   end
 
+  @doc """
+  Posts a QA test summary comment to VCS for the given QA run.
+  """
+  def post_vcs_test_summary(qa_run) do
+    qa_run =
+      Repo.preload(qa_run,
+        app_build: [preview: [project: :account]],
+        run_steps: :screenshots
+      )
+
+    preview = qa_run.app_build.preview
+    project = preview.project
+    comment_body = render_qa_summary_comment_body(qa_run, project)
+
+    if qa_run.issue_comment_id do
+      VCS.update_comment(%{
+        repository_full_handle: project.vcs_repository_full_handle,
+        comment_id: qa_run.issue_comment_id,
+        body: comment_body,
+        project: project
+      })
+    else
+      VCS.create_comment(%{
+        repository_full_handle: project.vcs_repository_full_handle,
+        git_ref: preview.git_ref,
+        body: comment_body,
+        project: project
+      })
+    end
+  end
+
+  defp render_qa_summary_comment_body(qa_run, project) do
+    preview = qa_run.app_build.preview
+    preview_url = "#{Environment.app_url()}/#{project.account.name}/#{project.name}/previews/#{preview.id}"
+    git_remote_url_origin = Projects.get_repository_url(project)
+
+    EEx.eval_string(qa_summary_template(),
+      assigns: %{
+        summary: qa_run.summary,
+        run_steps: qa_run.run_steps,
+        app_url: Environment.app_url(),
+        account_handle: project.account.name,
+        project_handle: project.name,
+        qa_run_id: qa_run.id,
+        prompt: qa_run.prompt,
+        preview_url: preview_url,
+        preview_display_name: preview.display_name,
+        commit_sha: preview.git_commit_sha,
+        git_remote_url_origin: git_remote_url_origin
+      }
+    )
+  end
+
+  defp qa_summary_template do
+    """
+    ### 🤖 QA Test Summary
```

Comment:
example output: https://github.com/tuist/tuist/pull/7829#issuecomment-3164567042

Definitely can be improved upon, but feel like this is a relatively good start